backend/src/bundles/statistics/statistics.service.ts

```diff
@@ -0,0 +1,190 @@
+import {
+    type StatisticsRecord,
+    compareDateInDiapasonWithoutTime,
+    compareDatesWithoutTime,
+    StatisticsPeriods,
+} from 'shared/build/index.js';
+
+import { type ResumeShareService } from '../resume-share/resume-share.service.js';
+import { StatisticsDays } from './enums/statistics-days.enum.js';
+import {
+    type GetStatisticsResponseDto,
+    type ResumeShareDetailsGetResponseDto,
+} from './types/types.js';
+
+class StatisticsService {
+    private resumeShareService: ResumeShareService;
+
+    public constructor(resumeShareService: ResumeShareService) {
+        this.resumeShareService = resumeShareService;
+    }
+
+    private getStatisticsHandleWeek(
+        data: ResumeShareDetailsGetResponseDto[],
+    ): GetStatisticsResponseDto {
+        const statistics: StatisticsRecord[] = [];
+
+        const dateWeekAgo = new Date();
+        dateWeekAgo.setDate(dateWeekAgo.getDate() - StatisticsDays.WEEK);
+
+        const accesses = data.flatMap((resume) => resume.accesses);
+
+        for (
+            const day = dateWeekAgo;
+            day <= new Date();
+            day.setDate(day.getDate() + 1)
+        ) {
+            statistics.push([
+                day.toLocaleDateString('en-US', {
+                    month: 'short',
+                    day: 'numeric',
+                }),
+                accesses.filter((access) => {
+                    return compareDatesWithoutTime(
+                        new Date(access.resumeShareAccessTime),
+                        day,
+                    );
+                }).length,
+            ]);
+        }
+
+        return {
+            data: statistics,
+            sum: statistics.reduce(
+                (accumulator, currentValue) => accumulator + currentValue[1],
+                0,
+            ),
+        };
```

Comment:
There is a simpler way to achieve it. Check date-fns package to work with dates
Same for other places where dates are used.

Reply:
Done